conan/tools/cmake/toolchain/toolchain.py

```diff
@@ -60,17 +60,20 @@ class CMakeToolchain(object):
         {% macro iterate_configs(var_config, action) %}
             {% for it, values in var_config.items() %}
                 {% set genexpr = namespace(str='') %}
+                {% for conf, value in values -%}
+                set(CONAN_DEF_{{ conf }}{{ it }} "{{ value }}")
+                {% endfor %}
                 {% for conf, value in values -%}
                     {% set genexpr.str = genexpr.str +
-                                          '$<IF:$<CONFIG:' + conf + '>,' + value|string + ',' %}
+                                          '$<IF:$<CONFIG:' + conf + '>,${CONAN_DEF_' + conf|string + it|string + '},' %}
                     {% if loop.last %}{% set genexpr.str = genexpr.str + '""' -%}{%- endif -%}
                 {% endfor %}
                 {% for i in range(values|count) %}{% set genexpr.str = genexpr.str + '>' %}
                 {% endfor %}
                 {% if action=='set' %}
                 set({{ it }} {{ genexpr.str }} CACHE STRING
                     "Variable {{ it }} conan-toolchain defined")
-                {% elif action=='add_compile_definitions' %}
+                {% elif action=='add_compile_definitions' -%}
```

Comment:
The case below is not quoting the ``add_compile_definitions()`` as in the other cases, not sure if intentional, or it is missing something (is the part that sets values per Debug/Release config)